src/viam/sdk/module/service.cpp

```diff
@@ -194,7 +195,7 @@ ::grpc::Status ModuleService_::RemoveResource(
     }
 
     try {
-        res->stop();
+        Stoppable::stop_if_stoppable(res);
```

Comment:
Here and above, in the past, if the implementation object inside a module had an `override` of `Resource::stop`, it'd get called here. Now, it won't, unless the implementation object explicitly derives from `Stoppable`. That seems like a breaking change for modules.

Reply:
What I'm trying to get at is that if you have an existing resource implemented (presumably as a module), it almost certainly has a `stop` method defined, because it had to: `Resource::stop` was pure virtual so you had to provide a definition.

And, that definition had paths that called it: from `ModuleService::reconfigureResource` if a non-`std::exception` exception was thrown, and from `ModuleService::RemoveResource`. And the steps taken in that `stop` may have been important.

After this change, the resource defined in the module will still have a `stop` method, but it won't be `Resource::stop` and it won't be `Stoppable::stop`: it will just be the `stop` method of the resource. None of the management code here can ever cause that `stop` to be invoked.

So it is something of a breaking change, in that if the implementation of your module was relying on `stop` in order to manage shutdown, it won't be getting called anymore. As an example, you needed to make changes to the TFLite module here.

Reply:
>So it is something of a breaking change, in that if the implementation of your module was relying on stop in order to manage shutdown, it won't be getting called anymore. As an example, you needed to make changes to the TFLite module here.

Yeah it's a good point, and after discussing offline, I think we should have `MLModelServiceTFLite : public Stoppable` for now despite `Stoppable` now modeling both gRPC `stop` methods and `stop` (acting as a "cleanup" method) for modular resources.